### DISCUSSION TITLE
Bun v1.0.1

### DISCUSSION THREAD

**Discussion by Jarred-Sumner** [oven-sh/bun#5117] [Announcements]
To install Bun v1.0.1:
```bash
curl https://bun.sh/install | bash
# or you can use npm
# npm install -g bun
```

To upgrade to Bun v1.0.1:
```bash
bun upgrade
```

**[Read Bun v1.0.1's release notes on Bun's blog](https://bun.sh/blog/bun-v1.0.1)**

## What's Changed
* docs: Fix typo in development.md by @TiranexDev in https://github.com/oven-sh/bun/pull/4578
* Fix `Bun.CryptoHasher` missing argument in docs by @samualtnorman in https://github.com/oven-sh/bun/pull/4585
* Make the link to application clickable in the terminal by @brablc in https://github.com/oven-sh/bun/pull/4544
* Move "Importing CJS from CJS" back out of low-level details section by @dfabulich in https://github.com/oven-sh/bun/pull/4600
* docs: fix `--backend` options list on cache.md by @vitordino in https://github.com/oven-sh/bun/pull/4599
* Fix code typo in transpiler.md by @fdb in https://github.com/oven-sh/bun/pull/4604
* docs(templates.md): spelling update by @rajatdua in https://github.com/oven-sh/bun/pull/4614
* Update Svelte Kit Docs by @e253 in https://github.com/oven-sh/bun/pull/4541
* Update mongoose.md by @tk120404 in https://github.com/oven-sh/bun/pull/4534
* Delete docs/ecosystem/buchta.md by @xxxhussein in https://github.com/oven-sh/bun/pull/4536
* fix(docs): fix "Connect to a WebSocket server" repeated documentation by @BeeMargarida in https://github.com/oven-sh/bun/pull/4615
* Fix listen() using unix socket if argument is a valid port by @Electroid in https://github.com/oven-sh/bun/pull/4620
* fix typo by @akumarujon in https://github.com/oven-sh/bun/pull/4639
* fix: remove `.only` from `bun-server.test.ts` test case by @chrishutchinson in https://github.com/oven-sh/bun/pull/4709
* fix(docs): fix wasi-js link by @wobsoriano in https://github.com/oven-sh/bun/pull/4741
* fix(docs); typo on Nuxt guide page by @thunfisch987 in https://github.com/oven-sh/bun/pull/4745
* Fix mongoose ecosystem guide by @Fecony in https://github.com/oven-sh/bun/pull/4740
* docs: fix spelling by @therealrinku in https://github.com/oven-sh/bun/pull/4763
* fix(bindings): add missing binding by @WingLim in https://github.com/oven-sh/bun/pull/4682
* docs: Update Bun.write(path, Response) example to be clear that it writes the body by @tom-sherman in https://github.com/oven-sh/bun/pull/4802
* Docs: Add compatibility page link to index.md by @Slikon in https://github.com/oven-sh/bun/pull/4809
* Update todo-tests.md by @toneyzhen in https://github.com/oven-sh/bun/pull/4771
* docs: remove duplicate sentence from debugger by @eknowles in https://github.com/oven-sh/bun/pull/4807
* Update modules.md - Explain case-insensitivity by @jeroenvanrensen in https://github.com/oven-sh/bun/pull/4782
* Replace unnecessary backticks (`) with double quotes (") in http doc code snippets by @diogo405 in https://github.com/oven-sh/bun/pull/4792
* Fixes #4588 by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/4804
* Support named imports for json & toml files at runtime by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/4783
* added .idea to init's .gitignore by @kpracuk in https://github.com/oven-sh/bun/pull/4951
* docs: changes wrong command name in the testing chapter by @AvantaR in https://github.com/oven-sh/bun/pull/4965
* Make detect-libc package match the upstream api by @fehnomenal in https://github.com/oven-sh/bun/pull/4910
* Fix punctuation by @matthewyu01 in https://github.com/oven-sh/bun/pull/4870
* Update http.js - Fix Methods Fallback naming by @Yonben in https://github.com/oven-sh/bun/pull/4948
* Fixed inconsistent log messages in the Quickstart guide by @pesterev in https://github.com/oven-sh/bun/pull/4942
* Remove unnecessary backticks in quickstart doc page by @diogo405 in https://github.com/oven-sh/bun/pull/4927
* Fix error message typo in js_ast.zig by @benjervis in https://github.com/oven-sh/bun/pull/4937
* fix(bun-lambda): Don't drop arguments in console.log(...) by @booniepepper in https://github.com/oven-sh/bun/pull/4992
* fix loading env from `.env.production` and friends by @dylan-conway in https://github.com/oven-sh/bun/pull/4630
* fix(BunFile) .slice offset on macOS by @cirospaciari in https://github.com/oven-sh/bun/pull/4991
* Fix `Buffer.from` to handle double-byte hex encoding strings by @Hanaasagi in https://github.com/oven-sh/bun/pull/4933
* fix(path): Fix edge case in `path.relative` by @Hanaasagi in https://github.com/oven-sh/bun/pull/4811
* Update gzip.md by @ebidel in https://github.com/oven-sh/bun/pull/5041
* add NODE_TLS_REJECT_UNAUTHORIZED for fetch/install/upgrade/create by @cirospaciari in https://github.com/oven-sh/bun/pull/4829
* fix(JSC): copy string in `toJSStringValueGC` by @Hanaasagi in https://github.com/oven-sh/bun/pull/5067
* Fix some bugs blocking Turborepo from using `bun run` by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/5071
* fix(node/path): Prevent memory corruption in parse by @davidmhewitt in https://github.com/oven-sh/bun/pull/5083
* fix up with `prettier` by @alexlamsl in https://github.com/oven-sh/bun/pull/5092

## New Contributors
* @samualtnorman made their first contribution in https://github.com/oven-sh/bun/pull/4585
* @brablc made their first contribution in https://github.com/oven-sh/bun/pull/4544
* @dfabulich made their first contribution in https://github.com/oven-sh/bun/pull/4600
* @vitordino made their first contribution in https://github.com/oven-sh/bun/pull/4599
* @fdb made their first contribution in https://github.com/oven-sh/bun/pull/4604
* @rajatdua made their first contribution in https://github.com/oven-sh/bun/pull/4614
* @e253 made their first contribution in https://github.com/oven-sh/bun/pull/4541
* @tk120404 made their first contribution in https://github.com/oven-sh/bun/pull/4534
* @BeeMargarida made their first contribution in https://github.com/oven-sh/bun/pull/4615
* @akumarujon made their first contribution in https://github.com/oven-sh/bun/pull/4639
* @chrishutchinson made their first contribution in https://github.com/oven-sh/bun/pull/4709
* @thunfisch987 made their first contribution in https://github.com/oven-sh/bun/pull/4745
* @Fecony made their first contribution in https://github.com/oven-sh/bun/pull/4740
* @therealrinku made their first contribution in https://github.com/oven-sh/bun/pull/4763
* @WingLim made their first contribution in https://github.com/oven-sh/bun/pull/4682
* @tom-sherman made their first contribution in https://github.com/oven-sh/bun/pull/4802
* @Slikon made their first contribution in https://github.com/oven-sh/bun/pull/4809
* @toneyzhen made their first contribution in https://github.com/oven-sh/bun/pull/4771
* @eknowles made their first contribution in https://github.com/oven-sh/bun/pull/4807
* @jeroenvanrensen made their first contribution in https://github.com/oven-sh/bun/pull/4782
* @diogo405 made their first contribution in https://github.com/oven-sh/bun/pull/4792
* @kpracuk made their first contribution in https://github.com/oven-sh/bun/pull/4951
* @AvantaR made their first contribution in https://github.com/oven-sh/bun/pull/4965
* @fehnomenal made their first contribution in https://github.com/oven-sh/bun/pull/4910
* @matthewyu01 made their first contribution in https://github.com/oven-sh/bun/pull/4870
* @Yonben made their first contribution in https://github.com/oven-sh/bun/pull/4948
* @pesterev made their first contribution in https://github.com/oven-sh/bun/pull/4942
* @benjervis made their first contribution in https://github.com/oven-sh/bun/pull/4937
* @booniepepper made their first contribution in https://github.com/oven-sh/bun/pull/4992
* @ebidel made their first contribution in https://github.com/oven-sh/bun/pull/5041
* @davidmhewitt made their first contribution in https://github.com/oven-sh/bun/pull/5083

**Full Changelog**: https://github.com/oven-sh/bun/compare/bun-v1.0.0...bun-v1.0.1

<hr /><em>This discussion was created from the release <a href='https://github.com/oven-sh/bun/releases/tag/bun-v1.0.1'>Bun v1.0.1</a>.</em>

**Comment by notxenonbox** [comment]
~~Great news! I am so happy for this project. However I am wondering whether or not Bun follows [Semantic Versioning](https://semver.org/). If so, based on my understanding, the addition of named imports for json and toml files, means this version would typically be named 1.1.0.~~

Never mind, I was not aware that this addition was a bug fix, I believe this should be mentioned in the announcement to avoid future misunderstandings.
